src/test/java/com/uber/jenkins/phabricator/conduit/DifferentialTest.java

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.uber.jenkins.phabricator.conduit;
+
+
+import hudson.EnvVars;
+import junit.framework.TestCase;
+import net.sf.json.JSONObject;
+import net.sf.json.groovy.JsonSlurper;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class DifferentialTest extends TestCase {
+    private static final String fakeDiffID = "not-a-real-id";
+    @Rule
+    Differential d;
+
+    public DifferentialTest() throws InterruptedException, ArcanistUsageException, IOException {
+        d = diffWithResponse();
+    }
+
+    @Test
+    public void testFetchRevisionID() throws Exception {
+        assertEquals(fakeDiffID, d.getRevisionID(false));
+    }
+
+    @Test
+    public void testGetPhabricatorLink() throws Exception {
+        assertTrue(d.getPhabricatorLink("http://example.com").contains(fakeDiffID));
+    }
+
+    @Test
+    public void testGetPhabricatorLinkInvalidURL() throws Exception {
+        // Try our best to join URLs, even when they are wrong
+        assertTrue(d.getPhabricatorLink("aoeu").contains("aoeu"));
+    }
+
+    @Test
+    public void testGetBuildStartedMessage() throws Exception {
+        assertTrue(d.getBuildStartedMessage(new EnvVars()).contains("Build started"));
+    }
+
+    private Differential diffWithResponse() throws InterruptedException, ArcanistUsageException, IOException {
+        DifferentialClient client = mock(DifferentialClient.class);
```

Comment:
At some point, we're probably going to add a MockDifferentialClient class to the test package and have some sort of response queue object in there that returns expected json blobs when we do callConduit. But separating the differential client and differential model is a HUGE step! This is some sweet stuff!! :)